action.yml

```diff
@@ -0,0 +1,50 @@
+name: Sync Salmon Directory
+description: Syncs a directory to a Google Cloud bucket using rclone.
+author: 'deephaven'
+inputs:
+  source:
+    required: true
+    type: string
+    description: 'The source directory to sync.'
+  destination:
+    required: true
+    type: string
+    description: 'The destination directory to sync. Relative to the bucket.'
+  project_number:
+    required: true
+    type: string
+    description: 'The Google Cloud project number.'
+  bucket:
+    required: true
+    type: string
+    description: 'The Google Cloud bucket to sync to.'
+  credentials:
+    required: true
+    type: string
+    description: 'The Google Cloud credentials.'
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup rclone
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y rclone=1.53.3-4ubuntu1.22.04.2
```

Comment:
Curious if using the rsync docker image as the base would be faster here. Not sure how long the update/install takes to run

Reply:
I'm not sure either, although thought the docker image would likely be a bit slower because there's more overhead. Not sure if it's something worth testing for a small difference at this time though.
This reminds me though that I should at least look into caching the rclone dependency so I'll do that.

Reply:
Probably doesn't matter since it does only take around 10-15s. I just remember something in the e2e tests for web-client-ui taking a decent amount of time because it does an apt-get update and install and it's slower than intended.

The other potential pitfall of this is when GitHub updates the runners to newer Ubuntu versions or if Ubuntu updates the upstream repos to remove the specific rclone version. Ubuntu 24 has rclone 1.60 and doesn't look like 1.53 is available

Reply:
this originated from https://github.com/deephaven/deephaven.io/pull/4394#discussion_r1778925564 so tagging in
@mofojed for thoughts
I suppose it is more likely that this rclone version is removed than rclone breaks something? But either way there is risk.

Reply:
It looks like the linked setup-rclone action on the marketplace downloads rclone from their repo releases rather than apt.

Also, it looks like reusable actions don't get to set their runner environment, so if we are downloading from apt we implicitly add a dependency that all of our actions which use this action must run on Ubuntu 22.04. Otherwise the apt install will fail.

So we probably need to remove apt install and either install directly from GitHub releases or use the docker base images to get a stable environment that won't be affected if we run this action in Ubuntu 24.04 or 20.04. GitHub currently supports Ubuntu 20, 22, and 24 releases for actions from what I see.

Reply:
Good points, and I think this has evolved to enough complexity where I should just use the setup-rclone action to handle this rather than reimplementing these behaviors.

Reply:
moved to the action